src/Common/src/System/Security/Cryptography/DerSequenceReader.cs

```diff
@@ -260,16 +284,28 @@ internal string ReadIA5String()
 
         internal DateTime ReadUtcTime()
         {
-            EatTag(DerTag.UTCTime);
+            return ReadTime(DerTag.UTCTime, "yyMMddHHmmss'Z'");
+        }
+
+        internal DateTime ReadGeneralizedTime()
+        {
+            // Currently only supports reading times with no fractional seconds or time differentials
+            // as RFC 2630 doesn't allow these. In case this is done, the format string has to be parsed
+            // to follow rules on X.680 and X.690.
+            return ReadTime(DerTag.GeneralizedTime, "yyyyMMddHHmmss'Z'");
+        }
+
+        private DateTime ReadTime(DerTag timeTag, string formatString)
```

Comment:
Now that both GeneralizedTime and UtcTime are here, please remove the redundant code from X509Certificates.  

ReadGeneralizedOrUtcTime is probably a good name for what X509 needs.